trackintel/model/trips.py

```diff
@@ -1,19 +1,25 @@
-from trackintel.analysis.tracking_quality import temporal_tracking_quality
-from trackintel.io.postgis import write_trips_postgis
-from trackintel.io.file import write_trips_csv
-from trackintel.model.util import _copy_docstring, _register_trackintel_accessor
-import pandas as pd
 import geopandas as gpd
+import pandas as pd
 
 import trackintel as ti
+from trackintel.analysis.tracking_quality import temporal_tracking_quality
+from trackintel.io.file import write_trips_csv
+from trackintel.io.postgis import write_trips_postgis
+from trackintel.model.util import (
+    _copy_docstring,
+    _register_trackintel_accessor,
+    TrackintelBase,
+    TrackintelDataFrame,
+    TrackintelGeoDataFrameWithFallback,
+)
 
 
 @_register_trackintel_accessor("as_trips")
-class TripsAccessor(object):
+def trips(*args, **kwargs):
```

Comment:
So, the other option would be to make this a class, right? What would be the disadvantage of having this Trips class that handles the TripsDataFrame and TripsGeoDataFrame in the background?

Reply:
I just read your PR description above again, and just to clarify: My suggestion now would be to keep all the code you have with the two classes, but having a Trips class, such that users can use the construtor of Trips and will automatically get an object either of type TripsDataFrame or TripsGeoDataFrame

Reply:
Having a `Trip` class that automates the fallback would be great (if possible). 
It will also be great if we can re-initialize a `gdf` to trackintel trips with `Trip(gdf)` once the geometry is (re-)added,

Reply:
Okay, I changed the code such that `Trips` does pretty much the same thing as `trips` but as a class.

@hongyeehh I don't understand your comment about the automated fallback. Isn't it already automated?
Automatically upcasting if a geometry is added may be a bit hard, but calling `Trips(gdf)` to get `TripsGeoDataFrame` should be trivial.